Fix AWS cloudfront log parsing

ltflb-bgdi opened the pull request:
- Bug


## Proposed commit message

Fixed AWS cloudfront log parsing

Please explain:
- WHAT:
  - Completely refactored parsing by replacing grok with csv to better comply to the cloudfront standard log  format.
  - Fixed duration name in GROK csv parser, which was parsed into a _tmp field but never used.as not available.
  - Fixed x-forwarded-for handling:
    Make sure only real ip addresses are used. All malformed ip-addresses are listed in error.message without raising a pipeline error.
  - Moved cloudfront.content_type to http.response.mime_type, which is an ecs field.
  - Added `event.kind: pipeline_error` to error handler.

- WHY:
  - The fixes above are needed to reliably process cloudfront logs. We have observed a lot of errors and grok timeouts, which lead to congestion and performance issues on our elastic clusters. With the new parser in place, parsing works without issues. Even though the pipeline contains much more processors than the one using grok, we do not see any performance impact.
  - The current, csv based parser lays the foundation of adding support of cloudfront realtime logs in the future, since realtime log fields are configurable and thus using grok is way too complex.



## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

## How to test this PR locally

The PR contains sample logs of the problematic events.
To be tested with `elastic-package test pipeline --data-streams cloudfront_logs `

Note: Please note that the log producing the error.message is not added to the sample logs since it would let the
tests fail. See: https://github.com/elastic/elastic-package/issues/1392

## Related issues

-  Relates #10507

ltflb-bgdi commented:
/test

efd6 commented on packages/aws/data_stream/cloudfront_logs/elasticsearch/ingest_pipeline/default.yml:
```diff
@@ -21,51 +21,316 @@ processors:
+      lang: painless
+      source: >-
+        ctx._tmp.timestamp = ctx._tmp.date + 'T' + ctx._tmp.time;
+      if: ctx?._tmp?.date != null && ctx?._tmp?.time != null
```
```suggestion
      if: ctx._tmp?.date != null && ctx._tmp?.time != null
```
(similar change throughout: `ctx` is always non-null)

ltflb-bgdi replied:
Done

efd6 commented on packages/aws/data_stream/cloudfront_logs/elasticsearch/ingest_pipeline/default.yml:
```diff
@@ -21,51 +21,316 @@ processors:
+      ignore_failure: true
+  - append:
+      field: related.ip
+      value: "{{ source.ip }}"
```
```suggestion
      value: "{{{ source.ip }}}"
```
(throughout)

ltflb-bgdi replied:
Done

efd6 commented on packages/aws/data_stream/cloudfront_logs/elasticsearch/ingest_pipeline/default.yml:
```diff
@@ -21,51 +21,316 @@ processors:
+      description: In rare cases, the field contains hex encoded whitspaces (\20). We replace them by whitespaces again to not break parsing.
+      field: _tmp.x_forwarded_for
+      target_field: _tmp.forwarded_ip
+      pattern: "\\\\x20"
```
```suggestion
      pattern: '\\x20'
```

ltflb-bgdi replied:
Done. Note that AWS has analyzed this issue and considered it as a bug, with a fix expected to be rolled out within 4 weeks from now. We could think about removing this workaround already.

efd6 commented on packages/aws/data_stream/cloudfront_logs/elasticsearch/ingest_pipeline/default.yml:
```diff
@@ -21,51 +21,316 @@ processors:
+      separator: ',\s*'
+      ignore_missing: true
+  - convert:
+      description: Creates a target field containing valid ip's if ALL values are valid ip's.
```
Losing all IPs if one is invalid seems unpleasant. This in conjunction with the special case you have in the pipeline test config suggest that a painless script over this array to clean it up would be a good thing to have. The only test failure that I saw was a case where the forwarded IP was localhost:8080, so a host port split and a mapping from "localhost" to "127.0.0.1" should do.

ltflb-bgdi replied:
I completely agree that we should not loose valid addresses in case one them is invalid. I implemented it this way due to complexity.

Now I have rewritten it by using a script and it should almost do what you recommended. The only difference is that I ignore the host/port split and just look for a value starting with localhost. I guess this should do the job, since receiving host port is really an edge case.

As you can see, the current implementation is far from ideal. Since I did not find a simple way to check if a string is an ip, I used a regex with patterns copied from logstash. Please let me know if there is a better way of doing this, since I really don't like it at all.

I still add invalid addresses to error.message, but did not add a test log for it in order to not break the test.

ltflb-bgdi commented on packages/aws/changelog.yml:
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "2.23.0"
```
Done

ltflb-bgdi commented on packages/aws/manifest.yml:
```diff
@@ -1,7 +1,7 @@
 name: aws
 title: AWS
-version: 2.22.0
+version: 2.23.0
```
Done

ltflb-bgdi replied:
Updated again to 2.22.2

ltflb-bgdi commented on packages/aws/data_stream/cloudfront_logs/fields/ecs.yml:
Done

ltflb-bgdi commented on packages/aws/data_stream/cloudfront_logs/elasticsearch/ingest_pipeline/default.yml:
```diff
@@ -21,51 +21,316 @@ processors:
-  - drop: 
-      if: "ctx.event.original.startsWith('#')"
+  - drop:
+      if: ctx?.event?.original?.startsWith('#')
```
Done

ltflb-bgdi commented on packages/aws/data_stream/cloudfront_logs/_dev/test/pipeline/test-cloudfront.log:
```diff
@@ -1,11 +1,14 @@
 2019-12-04	21:02:31	LAX1	392	89.160.20.112	GET	d111111abcdef8.cloudfront.net	/index.html	200	-	Mozilla/5.0%20(Windows%20NT%2010.0;%20Win64;%20x64)%20AppleWebKit/537.36%20(KHTML,%20like%20Gecko)%20Chrome/78.0.3904.108%20Safari/537.36	-	-	Hit	SOX4xwn4XV6Q4rgb7XiVGOHms_BGlTAC4KyHmureZmBNrjGdRLiNIQ==	d111111abcdef8.cloudfront.net	https	23	0.001	-	TLSv1.2	ECDHE-RSA-AES128-GCM-SHA256	Hit	HTTP/2.0	-	-	1 ...
 2019-12-04	21:02:31	LAX1	392	2a02:cf40:add:4002:91f2:a9b2:e09a:6fc6	GET	d111111abcdef8.cloudfront.net	/index.html	200	-	Mozilla/5.0%20(Windows%20NT%2010.0;%20Win64;%20x64)%20AppleWebKit/537.36%20(KHTML,%20like%20Gecko)%20Chrome/78.0.3904.108%20Safari/537.36	-	-	Hit	k6WGMNkEzR5BEM_SaF47gjtX9zBDO2m349OY2an0QPEaUum1ZOLrow==	d111111abcdef8.cloudfront.net	https	23	0.000	-	TLSv1.2	ECDHE-RSA-AES128-GCM- ...
-2019-12-04	21:02:31	LAX1	392	89.160.20.112	GET	d111111abcdef8.cloudfront.net	/index.html	200	-	Mozilla/5.0%20(Windows%20NT%2010.0;%20Win64;%20x64)%20AppleWebKit/537.36%20(KHTML,%20like%20Gecko)%20Chrome/78.0.3904.108%20Safari/537.36	-	-	Hit	f37nTMVvnKvV2ZSvEsivup_c2kZ7VXzYdjC-GUQZ5qNs-89BlWazbw==	d111111abcdef8.cloudfront.net	https	23	0.001	-	TLSv1.2	ECDHE-RSA-AES128-GCM-SHA256	Hit	HTTP/2.0	-	-	1 ...
+2019-12-04	21:02:31	LAX1	392	89.160.20.112	GET	d111111abcdef8.cloudfront.net	/index.html	200	-	Mozilla/5.0%20(Windows%20NT%2010.0;%20Win64;%20x64)%20AppleWebKit/537.36%20(KHTML,%20like%20Gecko)%20Chrome/78.0.3904.108%20Safari/537.36	-	-	Hit	f37nTMVvnKvV2ZSvEsivup_c2kZ7VXzYdjC-GUQZ5qNs-89BlWazbw==	d111111abcdef8.cloudfront.net	https	23	0.001	-	TLSv1.2	ECDHE-RSA-AES128-GCM-SHA256	Hit	HTTP/2.0	-	-	1 ...
```
Trailing tabs should not have an effect on the csv parsers. Still, according to the AWS specs we should not see this.

efd6 replied:
If this doesn't break the test, can we leave it in?

ltflb-bgdi replied:
We could. But our vs-code setup is configured to automatically remove trailing spaces after editing.
I have done a test with trailing tabs by adding them with another editor and the parser did not fail.
Given that fact and that trailing tabs should not appear in real life, I vote for letting the trailing tab removed.

efd6 replied:
> But our vs-code setup is configured to automatically remove trailing spaces after editing.

I don't think you should do that here; there may be tests in other packages that _depend_ on trailing whitespace (note that we do not have an .editorconfig file in this repo so there is no guidance on the expectation of format).

While real data should not have the trailing tab, recent events have again demonstrated that testing with artifacts that match expectations can lead to bad outcomes. I would prefer that we show that we are robust to this case.

ltflb-bgdi commented on packages/aws/data_stream/cloudfront_logs/_dev/test/pipeline/test-cloudfront.log:
```diff
@@ -1,11 +1,14 @@
-#Version: 1.0
+#Version:1.0
```
No, it is not. I have changed it again. Since we drop all messages starting with #, it has no effect.

ltflb-bgdi commented on packages/aws/data_stream/cloudfront_logs/_dev/test/pipeline/test-cloudfront.log:
```diff
@@ -1,11 +1,14 @@
-2022-04-19 12:29:36 SEA19-C2 10157 81.2.69.143 POST d111111abcdef8.cloudfront.net /getApplications 200 https://test.com/global Mozilla/5.0%20(Windows%20NT%2010.0;%20Win64;%20x64)%20AppleWebKit/537.36%20(KHTML,%20like%20Gecko)%20Chrome/100.0.4896.127%20Safari/537.36 source=global - Miss hrsHM5OM6sTIXUleC1G20YtDxMf5Cq0Jbz0pwhVpod2kgEn_W6akCQ== test.com https 1057 0.238 - TLSv1.3 TLS_AES_128_GCM_SHA ...
-2022-04-19 12:29:36 SEA19-C2 10157 81.2.69.143 POST d111111abcdef8.cloudfront.net /getApplications 000 https://test.com/global Mozilla/5.0%20(Windows%20NT%2010.0;%20Win64;%20x64)%20AppleWebKit/537.36%20(KHTML,%20like%20Gecko)%20Chrome/100.0.4896.127%20Safari/537.36 source=global - Miss hrsHM5OM6sTIXUleC1G20YtDxMf5Cq0Jbz0pwhVpod2kgEn_W6akCQ== test.com https 1057 0.238 - TLSv1.3 TLS_AES_128_GCM_SHA ...
-2022-11-15 08:43:04 SEA19-C2 10157 81.2.69.143 GET d111111abcdef8.cloudfront.net /getApplications 200 https://test.com/global Mozilla/5.0%20(X11;%20Linux%20x86_64)%20AppleWebKit/537.36%20(KHTML,%20like%20Gecko)%20HeadlessChrome/100.0.4896.88%20Safari/537.36 - - Miss hrsHM5OM6sTIXUleC1G20YtDxMf5Cq0Jbz0pwhVpod2kgEn_W6akCQ== test.com https 1057 0.093 81.2.69.142,216.160.83.56 TLSv1.2 ECDHE-RSA-AES12 ...
+2022-04-19	12:29:36	SEA19-C2	10157	81.2.69.143	POST	d111111abcdef8.cloudfront.net	/getApplications	200	https://test.com/global	Mozilla/5.0%20(Windows%20NT%2010.0;%20Win64;%20x64)%20AppleWebKit/537.36%20(KHTML,%20like%20Gecko)%20Chrome/100.0.4896.127%20Safari/537.36	source=global	-	Miss	hrsHM5OM6sTIXUleC1G20YtDxMf5Cq0Jbz0pwhVpod2kgEn_W6akCQ==	test.com	https	1057	0.238	-	TLSv1.3	TLS_AES_128_GCM_SHA ...
```
Yes they do, since they contain space separators instead of tab. Since `\t` accepts both, it worked in the past.
According to the aws docs, values are tab separated. Thus relying on tabs only is save/better IMO.

I'll add a comment with a link to the docs later on.

efd6 replied:
If this doesn't break the test, can we leave it in?

anil-elastic replied:
Please add below test case also ( [refernce ](https://github.com/elastic/integrations/issues/10507) ) :

2024-07-13	15:29:45	EWR53-C1	198083	127.0.0.1	GET	xxxxxxxxxxxxx.cloudfront.net	/en(test)	404	https://domain.tld/	User-Agent:%20Mozilla/4.0%20(compatible;%20MSIE%207.0;%20Windows%20NT%205.1;%20360SE)	-	-	Error	somevalidbase64==	domain.tld	https	609	0.318	-	TLSv1.3	TLS_AES_128_GCM_SHA256	Error	HTTP/1.1	-	-	50294	0.318	Error	text/html	-

ltflb-bgdi commented on packages/aws/data_stream/cloudfront_logs/_dev/test/pipeline/test-common-config.yml:
```diff
@@ -1,3 +1,6 @@
 fields:
   tags:
     - preserve_original_event
+    - elastic_package_remove_error_message # Used to solve elastic-package test error on adding error message to event.
+dynamic_fields:
+  'event.ingested': "^.*$"
```
Done

ltflb-bgdi replied:
I really don't like the testing-hack either! And to be honest, I really don't get it why there is no fix from elastic yet to allow the `error.message` field in the test, since is is just a common ecs field, which even can be part of the original message. This is a known problem with an existing [issue](https://github.com/elastic/elastic-package/issues/1392) waiting to be fixed for almost a year.

efd6 commented:
/test

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/14391)
* Commit: ea0a6e705d97c2979207564380f790765cca4d78

### Failed CI Steps
* [:pipeline::arrow_up: Upload Pipeline: .buildkite/pipeline.yml](https://buildkite.com/elastic/integrations/builds/14391#019129c5-5038-4241-b023-f935fc1b02ff)

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/14391","number":14391,"commit":"ea0a6e705d97c2979207564380f790765cca4d78"}],"number":14391}
buildkite-pr-comment-->

efd6 commented:
The system tests will need to be updated to use tabs as well. That file is [here](https://github.com/elastic/integrations/blob/main/packages/aws/data_stream/cloudfront_logs/_dev/deploy/tf/files/test-cloudfront.log).

ltflb-bgdi commented:
/test

ltflb-bgdi commented:
/test

ltflb-bgdi commented:
/test

ltflb-bgdi commented:
/test

efd6 commented:
/test

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/14445)
* Commit: 945386b69b19304572b4f8e1aec6aa4481a053cb

### Failed CI Steps
* [Check integrations aws](https://buildkite.com/elastic/integrations/builds/14445#01912eae-1af8-4862-91c5-75e01b819153)

### History
* :broken_heart: [Build #14391](https://buildkite.com/elastic/integrations/builds/14391) failed ea0a6e705d97c2979207564380f790765cca4d78

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/14445","number":14445,"commit":"945386b69b19304572b4f8e1aec6aa4481a053cb"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/14391","number":14391,"commit":"ea0a6e705d97c2979207564380f790765cca4d78"}],"number":14445}
buildkite-pr-comment-->

efd6 commented:
CI fails with:
```
test case failed: one or more errors found in document: [0] field "aws.cloudfront.content_type" is undefined
```
and I see the following failure locally:
```
╭─────────┬─────────────────┬───────────┬───────────┬────────────────────────────────────────────────────────────────┬─────────────────╮
│ PACKAGE │ DATA STREAM     │ TEST TYPE │ TEST NAME │ RESULT                                                         │    TIME ELAPSED │
├─────────┼─────────────────┼───────────┼───────────┼────────────────────────────────────────────────────────────────┼─────────────────┤
│ aws     │ cloudfront_logs │ system    │ default   │ FAIL: observed hit count 11 did not match expected hit count 9 │ 2m42.650776006s │
╰─────────┴─────────────────┴───────────┴───────────┴────────────────────────────────────────────────────────────────┴─────────────────╯
```

ltflb-bgdi commented:
/test

efd6 commented:
/test

ltflb-bgdi commented:
> CI fails with:
> 
> ```
> test case failed: one or more errors found in document: [0] field "aws.cloudfront.content_type" is undefined
> ```
> 
> and I see the following failure locally:
> 
> ```
> ╭─────────┬─────────────────┬───────────┬───────────┬────────────────────────────────────────────────────────────────┬─────────────────╮
> │ PACKAGE │ DATA STREAM     │ TEST TYPE │ TEST NAME │ RESULT                                                         │    TIME ELAPSED │
> ├─────────┼─────────────────┼───────────┼───────────┼────────────────────────────────────────────────────────────────┼─────────────────┤
> │ aws     │ cloudfront_logs │ system    │ default   │ FAIL: observed hit count 11 did not match expected hit count 9 │ 2m42.650776006s │
> ╰─────────┴─────────────────┴───────────┴───────────┴────────────────────────────────────────────────────────────────┴─────────────────╯
> ```

This is strange. Local testing works for me and the "aws.cloudfront.content_type" field does not exist anymore. It is replaced with "http.response.mime_type".

I have changed the expected count to 11 (We have two more logs).

Please let me know  if it works now.

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/14468)
* Commit: 646cdcc2b28e5095ebfcc305f75478b730066a0a

### Failed CI Steps
* [Check integrations aws](https://buildkite.com/elastic/integrations/builds/14468#01913153-38cf-4bfa-a2da-360f4b89a6cd)

### History
* :broken_heart: [Build #14445](https://buildkite.com/elastic/integrations/builds/14445) failed 945386b69b19304572b4f8e1aec6aa4481a053cb
* :broken_heart: [Build #14391](https://buildkite.com/elastic/integrations/builds/14391) failed ea0a6e705d97c2979207564380f790765cca4d78

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/14468","number":14468,"commit":"646cdcc2b28e5095ebfcc305f75478b730066a0a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/14445","number":14445,"commit":"945386b69b19304572b4f8e1aec6aa4481a053cb"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/14391","number":14391,"commit":"ea0a6e705d97c2979207564380f790765cca4d78"}],"number":14468}
buildkite-pr-comment-->

ltflb-bgdi commented:
/test

efd6 commented:
/test

elasticmachine commented:
### :rocket: Benchmarks report


#### Package `aws` :&#43;1:(8) :green_heart:(2) :broken_heart:(7)
<details>
<summary>Expand to view</summary>

Data stream | Previous EPS | New EPS | Diff (%) | Result
----------- | ------------ | ------- | -------- | ------
`route53_public_logs` | 10416.67 | 7518.8 | -2897.87 (-27.82%) | :broken_heart:
`s3access` | 4950.5 | 3154.57 | -1795.93 (-36.28%) | :broken_heart:
`securityhub_findings` | 1085.78 | 871.84 | -213.94 (-19.7%) | :broken_heart:
`cloudwatch_logs` | 500000 | 333333.33 | -166666.67 (-33.33%) | :broken_heart:
`elb_logs` | 5617.98 | 3875.97 | -1742.01 (-31.01%) | :broken_heart:
`emr_logs` | 13157.89 | 8196.72 | -4961.17 (-37.7%) | :broken_heart:
`firewall_logs` | 2538.07 | 2114.16 | -423.91 (-16.7%) | :broken_heart:
</details>


To see the full report comment with `/test benchmark fullreport`
<!--COMMENT_GENERATED_WITH_ID_benchmark-report-->

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/14499)
* Commit: 5853e6b8108532ed4bddc0caac10f8ca173f4be4

### History
* :broken_heart: [Build #14468](https://buildkite.com/elastic/integrations/builds/14468) failed 646cdcc2b28e5095ebfcc305f75478b730066a0a
* :broken_heart: [Build #14445](https://buildkite.com/elastic/integrations/builds/14445) failed 945386b69b19304572b4f8e1aec6aa4481a053cb
* :broken_heart: [Build #14391](https://buildkite.com/elastic/integrations/builds/14391) failed ea0a6e705d97c2979207564380f790765cca4d78

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/14499","number":14499,"commit":"5853e6b8108532ed4bddc0caac10f8ca173f4be4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/14468","number":14468,"commit":"646cdcc2b28e5095ebfcc305f75478b730066a0a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/14445","number":14445,"commit":"945386b69b19304572b4f8e1aec6aa4481a053cb"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/14391","number":14391,"commit":"ea0a6e705d97c2979207564380f790765cca4d78"}],"number":14499}
buildkite-pr-comment-->

ltflb-bgdi commented:
/test benchmark fullreport

ltflb-bgdi commented:
@efd6 Please check if it works now as expected. If not, please take over since it is getting hard for me to deal with stuff I can not test by myself and half a day of time shift. 

@andrewkroh could you step in since our time-zones are more overlapping. Thanks

elasticmachine commented:
Pinging @elastic/security-service-integrations (Team:Security-Service Integrations)

andrewkroh commented:
/test

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/14527)
* Commit: 0202a5986b8a215469d21333719ff68a7087a016

### Failed CI Steps
* [Check integrations aws](https://buildkite.com/elastic/integrations/builds/14527#0191389a-db76-4e21-ac2f-48783fc0137b)

### History
* :green_heart: [Build #14499](https://buildkite.com/elastic/integrations/builds/14499) succeeded 5853e6b8108532ed4bddc0caac10f8ca173f4be4
* :broken_heart: [Build #14468](https://buildkite.com/elastic/integrations/builds/14468) failed 646cdcc2b28e5095ebfcc305f75478b730066a0a
* :broken_heart: [Build #14445](https://buildkite.com/elastic/integrations/builds/14445) failed 945386b69b19304572b4f8e1aec6aa4481a053cb
* :broken_heart: [Build #14391](https://buildkite.com/elastic/integrations/builds/14391) failed ea0a6e705d97c2979207564380f790765cca4d78

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/14527","number":14527,"commit":"0202a5986b8a215469d21333719ff68a7087a016"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/14499","number":14499,"commit":"5853e6b8108532ed4bddc0caac10f8ca173f4be4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/14468","number":14468,"commit":"646cdcc2b28e5095ebfcc305f75478b730066a0a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/14445","number":14445,"commit":"945386b69b19304572b4f8e1aec6aa4481a053cb"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/14391","number":14391,"commit":"ea0a6e705d97c2979207564380f790765cca4d78"}],"number":14527}
buildkite-pr-comment-->

andrewkroh commented on packages/aws/data_stream/cloudfront_logs/elasticsearch/ingest_pipeline/default.yml:
```diff
@@ -174,25 +430,21 @@ processors:
+      value: pipeline_error
+  - append:
+      field: error.message
+      value: '{{{ _ingest.on_failure_message }}}'
```
```suggestion
      value: >-
        Processor '{{ _ingest.on_failure_processor_type }}'
        {{#_ingest.on_failure_processor_tag}}with tag '{{ _ingest.on_failure_processor_tag }}'
        {{/_ingest.on_failure_processor_tag}}failed with message '{{ _ingest.on_failure_message }}'
```

andrewkroh commented:
The PR needs:

- `elastic-package build`
- `elastic-package test pipeline -g -d cloudfront_logs`

to update the generated docs expected.json files.

ltflb-bgdi commented:
> (I would do it, but this PR does not allow maintainers to push.)

Only users which are maintainers of our fork are allowed to push. We have created a specific team for elastic developers.
Currently the only member is `muthu-mps`. We could add you to the list as well if you want.

andrewkroh commented:
/test

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/14579)
* Commit: 73b251d91adffe1d9fc6c1f0aecefed67c36ab9e

### History
* :broken_heart: [Build #14527](https://buildkite.com/elastic/integrations/builds/14527) failed 0202a5986b8a215469d21333719ff68a7087a016
* :green_heart: [Build #14499](https://buildkite.com/elastic/integrations/builds/14499) succeeded 5853e6b8108532ed4bddc0caac10f8ca173f4be4
* :broken_heart: [Build #14468](https://buildkite.com/elastic/integrations/builds/14468) failed 646cdcc2b28e5095ebfcc305f75478b730066a0a
* :broken_heart: [Build #14445](https://buildkite.com/elastic/integrations/builds/14445) failed 945386b69b19304572b4f8e1aec6aa4481a053cb
* :broken_heart: [Build #14391](https://buildkite.com/elastic/integrations/builds/14391) failed ea0a6e705d97c2979207564380f790765cca4d78

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/14579","number":14579,"commit":"73b251d91adffe1d9fc6c1f0aecefed67c36ab9e"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/14527","number":14527,"commit":"0202a5986b8a215469d21333719ff68a7087a016"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/14499","number":14499,"commit":"5853e6b8108532ed4bddc0caac10f8ca173f4be4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/14468","number":14468,"commit":"646cdcc2b28e5095ebfcc305f75478b730066a0a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/14445","number":14445,"commit":"945386b69b19304572b4f8e1aec6aa4481a053cb"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/14391","number":14391,"commit":"ea0a6e705d97c2979207564380f790765cca4d78"}],"number":14579}
buildkite-pr-comment-->

ltflb-bgdi commented:
@andrewkroh @efd6 could you start the testing please?

Question: Is there a way contributors can start and analyze CI stuff as well?

ltflb-bgdi commented:
/test

andrewkroh commented:
> Question: Is there a way contributors can start and analyze CI stuff as well?

CI is setup to only trigger for elastic members for security reasons.

Are you using `elastic-package` as part of developing these changes? Locally you can run commands similar to what CI is doing, like

- `elastic-package check`
- `elastic-package test pipeline`
- `elastic-package test system`

Reference: https://www.elastic.co/guide/en/integrations-developer/current/elastic-package.html#_elastic_package_check

andrewkroh commented:
/test

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/14678)
* Commit: ecf13e436218244cd5b43e3068b1972741007762

### History
* :green_heart: [Build #14579](https://buildkite.com/elastic/integrations/builds/14579) succeeded 73b251d91adffe1d9fc6c1f0aecefed67c36ab9e
* :broken_heart: [Build #14527](https://buildkite.com/elastic/integrations/builds/14527) failed 0202a5986b8a215469d21333719ff68a7087a016
* :green_heart: [Build #14499](https://buildkite.com/elastic/integrations/builds/14499) succeeded 5853e6b8108532ed4bddc0caac10f8ca173f4be4
* :broken_heart: [Build #14468](https://buildkite.com/elastic/integrations/builds/14468) failed 646cdcc2b28e5095ebfcc305f75478b730066a0a
* :broken_heart: [Build #14445](https://buildkite.com/elastic/integrations/builds/14445) failed 945386b69b19304572b4f8e1aec6aa4481a053cb
* :broken_heart: [Build #14391](https://buildkite.com/elastic/integrations/builds/14391) failed ea0a6e705d97c2979207564380f790765cca4d78

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/14678","number":14678,"commit":"ecf13e436218244cd5b43e3068b1972741007762"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/14579","number":14579,"commit":"73b251d91adffe1d9fc6c1f0aecefed67c36ab9e"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/14527","number":14527,"commit":"0202a5986b8a215469d21333719ff68a7087a016"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/14499","number":14499,"commit":"5853e6b8108532ed4bddc0caac10f8ca173f4be4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/14468","number":14468,"commit":"646cdcc2b28e5095ebfcc305f75478b730066a0a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/14445","number":14445,"commit":"945386b69b19304572b4f8e1aec6aa4481a053cb"}],"number":14678}
buildkite-pr-comment-->

ltflb-bgdi commented:
> * elastic-package test system



> > Question: Is there a way contributors can start and analyze CI stuff as well?
> 
> CI is setup to only trigger for elastic members for security reasons.
> 
> Are you using `elastic-package` as part of developing these changes? Locally you can run commands similar to what CI is doing, like
> 
> * `elastic-package check`
> * `elastic-package test pipeline`
> * `elastic-package test system`
> 
> Reference: https://www.elastic.co/guide/en/integrations-developer/current/elastic-package.html#_elastic_package_check

I use `elastic-package check` and `elastic-package test pipeline` for my daily work. One thing I have just learned about `elastic-package check` is that it only does lint and build, but not format as mentioned in the docs.

The command output only mentions lint and build. To verify I have added some extra lines to the end of a fields.yml file.
Not removed with `elastic-package check`, removed with `elastic-package check`.

Finally, a look at the code shows this: https://github.com/elastic/elastic-package/blob/1412f01ca7895ee47681dc250a4d7acdd49da0ea/cmd/check.go#L26

Should contain setupFormatCommand(), as well IMO.

ltflb-bgdi commented:
@andrewkroh is there anything I can do to let the `quality gate` checks pass?

ishleenk17 commented:
/test

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/14803)
* Commit: f8b7aa8fc6df9125df5b5318f39edccd57636172

### Failed CI Steps
* [:pipeline::arrow_up: Upload Pipeline: .buildkite/pipeline.yml](https://buildkite.com/elastic/integrations/builds/14803#01917112-f330-4361-918b-ef860321f5a9)

### History
* :green_heart: [Build #14678](https://buildkite.com/elastic/integrations/builds/14678) succeeded ecf13e436218244cd5b43e3068b1972741007762
* :green_heart: [Build #14579](https://buildkite.com/elastic/integrations/builds/14579) succeeded 73b251d91adffe1d9fc6c1f0aecefed67c36ab9e
* :broken_heart: [Build #14527](https://buildkite.com/elastic/integrations/builds/14527) failed 0202a5986b8a215469d21333719ff68a7087a016
* :green_heart: [Build #14499](https://buildkite.com/elastic/integrations/builds/14499) succeeded 5853e6b8108532ed4bddc0caac10f8ca173f4be4
* :broken_heart: [Build #14468](https://buildkite.com/elastic/integrations/builds/14468) failed 646cdcc2b28e5095ebfcc305f75478b730066a0a
* :broken_heart: [Build #14445](https://buildkite.com/elastic/integrations/builds/14445) failed 945386b69b19304572b4f8e1aec6aa4481a053cb

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/14803","number":14803,"commit":"f8b7aa8fc6df9125df5b5318f39edccd57636172"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/14678","number":14678,"commit":"ecf13e436218244cd5b43e3068b1972741007762"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/14579","number":14579,"commit":"73b251d91adffe1d9fc6c1f0aecefed67c36ab9e"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/14527","number":14527,"commit":"0202a5986b8a215469d21333719ff68a7087a016"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/14499","number":14499,"commit":"5853e6b8108532ed4bddc0caac10f8ca173f4be4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/14468","number":14468,"commit":"646cdcc2b28e5095ebfcc305f75478b730066a0a"}],"number":14803}
buildkite-pr-comment-->

agithomas reviewed (approved):
LGTM!

andrewkroh commented:
/test

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/15117)
* Commit: c2db3bd5cd78610d8991de4eeda1fe2575957980

### Failed CI Steps
* [Check integrations aws](https://buildkite.com/elastic/integrations/builds/15117#019198bc-da49-4887-8316-4bb7aa633dc2)

### History
* :broken_heart: [Build #14803](https://buildkite.com/elastic/integrations/builds/14803) failed f8b7aa8fc6df9125df5b5318f39edccd57636172
* :green_heart: [Build #14678](https://buildkite.com/elastic/integrations/builds/14678) succeeded ecf13e436218244cd5b43e3068b1972741007762
* :green_heart: [Build #14579](https://buildkite.com/elastic/integrations/builds/14579) succeeded 73b251d91adffe1d9fc6c1f0aecefed67c36ab9e
* :broken_heart: [Build #14527](https://buildkite.com/elastic/integrations/builds/14527) failed 0202a5986b8a215469d21333719ff68a7087a016
* :green_heart: [Build #14499](https://buildkite.com/elastic/integrations/builds/14499) succeeded 5853e6b8108532ed4bddc0caac10f8ca173f4be4
* :broken_heart: [Build #14468](https://buildkite.com/elastic/integrations/builds/14468) failed 646cdcc2b28e5095ebfcc305f75478b730066a0a

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/15117","number":15117,"commit":"c2db3bd5cd78610d8991de4eeda1fe2575957980"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/14803","number":14803,"commit":"f8b7aa8fc6df9125df5b5318f39edccd57636172"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/14678","number":14678,"commit":"ecf13e436218244cd5b43e3068b1972741007762"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/14579","number":14579,"commit":"73b251d91adffe1d9fc6c1f0aecefed67c36ab9e"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/14527","number":14527,"commit":"0202a5986b8a215469d21333719ff68a7087a016"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/14499","number":14499,"commit":"5853e6b8108532ed4bddc0caac10f8ca173f4be4"}],"number":15117}
buildkite-pr-comment-->

andrewkroh commented:
Failure message:

```
<testcase name="system test: default" classname="aws.cloudfront_logs" time="95.697515564">
<failure>
observed hit count 12 did not match expected hit count 11
</failure>
</testcase>
```

andrewkroh commented on packages/aws/data_stream/cloudfront_logs/_dev/test/system/test-default-config.yml:
```diff
@@ -9,4 +9,4 @@ data_stream:
     preserve_original_event: true
 assert:
-  hit_count: 9
+  hit_count: 11
```
```suggestion
  hit_count: 12
```

I confirmed there are 12 logs in test-cloudfront.log (not counting header).

andrewkroh commented:
/test

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/15427)
* Commit: 682261ebd87ac8416e01aba6ba53f963b2c84cad

### History
* :broken_heart: [Build #15117](https://buildkite.com/elastic/integrations/builds/15117) failed c2db3bd5cd78610d8991de4eeda1fe2575957980
* :broken_heart: [Build #14803](https://buildkite.com/elastic/integrations/builds/14803) failed f8b7aa8fc6df9125df5b5318f39edccd57636172
* :green_heart: [Build #14678](https://buildkite.com/elastic/integrations/builds/14678) succeeded ecf13e436218244cd5b43e3068b1972741007762
* :green_heart: [Build #14579](https://buildkite.com/elastic/integrations/builds/14579) succeeded 73b251d91adffe1d9fc6c1f0aecefed67c36ab9e
* :broken_heart: [Build #14527](https://buildkite.com/elastic/integrations/builds/14527) failed 0202a5986b8a215469d21333719ff68a7087a016

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/15427","number":15427,"commit":"682261ebd87ac8416e01aba6ba53f963b2c84cad"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/15117","number":15117,"commit":"c2db3bd5cd78610d8991de4eeda1fe2575957980"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/14803","number":14803,"commit":"f8b7aa8fc6df9125df5b5318f39edccd57636172"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/14678","number":14678,"commit":"ecf13e436218244cd5b43e3068b1972741007762"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/14579","number":14579,"commit":"73b251d91adffe1d9fc6c1f0aecefed67c36ab9e"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/14527","number":14527,"commit":"0202a5986b8a215469d21333719ff68a7087a016"}],"number":15427}
buildkite-pr-comment-->

andrewkroh commented:
Everything looks in order to me now.

@elastic/obs-infraobs-integrations, as codeowner of cloudfront_logs, do you want to handle merging?

ltflb-bgdi commented:
@elastic/obs-infraobs-integrations Any update?

elasticmachine commented:
Package aws - 2.24.1 containing this change is available at https://epr.elastic.co/search?package=aws
